docs/DeveloperGuide.md

```diff
@@ -1,13 +1,413 @@
+| listExpenses()                         | void        | Lists out all expenses saved                                 |
+| listExpensesWithCategory(category)     | String      | Lists out all expenses saved with the specified category     |
+| listExpensesWithDate(month)            | String      | Lists out all expenses saved in specified month              |
+| listExpensesWithCategoryAndDate(month) | String      | Lists out all expenses saved in specified month and category |
+
+Then, with the variations of `listExpenses()` commands in `ExpenseManager`, the list of expenses will be displayed to 
+the user using the `Ui` class `displayToUser()` method.
+
+The following UML Sequence diagram shows how the Parser works to obtain the relevant inputs for the List Expense Feature
+![ListExpenseSequenceDiagram.drawio.png](diagrams/ListExpenseSequenceDiagram.drawio.png)
+
+#### 4.9 List Income Feature
+The List Income feature enables users to view saved income in the application. Additionally, user may add additional
+filters to display only desired months. The total income amount based on the displayed income will be summed and 
+displayed to the user. This feature is controlled by the `ListIncomeCommand` class, where it is initialized by the 
+`Parser` class. The `Parser` class uses `ListIncomeValidator` to check and extract the filter field that the user
+input. The `ListIncomeCommand` object is then created with a Category and YearMonth Attribute.
+
+| Class Attribute | Variable Type | Relevance                                            |
+|-----------------|---------------|------------------------------------------------------|
+| month           | YearMonth     | The specific month and year of expenses to be listed |
+
+The `BudgetBuddy` class then calls the `execute()` method of `ListIncomeCommand` object which uses the following
+methods in `IncomeManager` class to list expenses based on user input category and month. If user does not specify a
+specific month, it will show every valid expense instead.
+
+| Method                      | Return Type | Relevance                                                    |
+|-----------------------------|-------------|--------------------------------------------------------------|
+| listIncomes()               | void        | Lists out all expenses saved                                 |
+| listIncomesWithMonth(month) | String      | Lists out all expenses saved in specified month              |
+
+
+Then, with the variations of `listIncomes()` commands in `IncomeManager`, the list of incomes will be displayed to
+the user using the `Ui` class `displayToUser()` method.
+
+The following UML Sequence diagram shows how the Parser works to obtain the relevant inputs for the List Income Feature
+
+The following UML Sequence diagram shows how the Parser works to obtain the relevant inputs for the List Income Feature
+![ListIncomeSequenceDiagram.drawio.png](diagrams/ListIncomeSequenceDiagram.drawio.png)
```

Comment:
Should the constructor be called here for :ListIncomeCommand?
![image](https://github.com/user-attachments/assets/0e39c4bd-9d55-47c4-b179-715cb0365226)